src/values/ObjectValue.js

```diff
@@ -645,25 +645,46 @@ export default class ObjectValue extends ConcreteValue {
     }
 
     if (!(P instanceof AbstractValue)) return this.$Get(P, Receiver);
+
+    // A string coercion might have side-effects.
+    // We assume that simple objects mean that they don't have a side-effectful
+    // valueOf and toString. (Is this enforced?)
```

Comment:
No, a simple object does not have getters and setters and does not have symbol properties (and this applies to its prototype chain as well -- excepting the original Object.prototype object). We could expand this definition to mean that valueOf and toString will be well behaved, but that is non trivial. It seems that the expanded definition is so useful that it is already a de facto part of the definition, so this need to be made into an issue. While we are at it, we can also lift the restriction of symbol properties.

Reply:
Yea, it seems like this is already the case in many other places. Even in this file it was already the case. Since it is in many place I don't really want to change that definition in this PR but maybe a follow up. I'll change this to a todo with an issue.

It seems difficult to guarantee that simple objects don't have valueOf/toString though. E.g. Object.assign can return a simple partial object I think but how does it guarantee that there are no keys with valueOf and toString?